pyfr/plugins/base.py

```diff
@@ -161,3 +165,87 @@ def _prepare_region_data_eset(self, intg, eset):
 
             self._ele_regions.append((doff, etype, darr))
             self._ele_region_data[f'{etype}_idxs'] = darr
+
+    def prepare_surface_data(self, intg):
+        if self._rgn:
+            self._surf_region_data = self._rgn.surface_faces(intg.system.mesh,
+                                                             intg.rallocs)
+
+class ElementMixin:
+    def __init__(self, intg, *args, **kwargs):
+        super().__init__(intg, *args, **kwargs)
+
+        self._surf_quad = {}
+
+    def surf_quad(self, itype, proj, *args):
+        try:
+            ppts, pwts = self._surf_quad[itype, proj]
+        except KeyError:
+            # Obtain quadrature info
+            rname = self.cfg.get(f'solver-interfaces-{itype}', 'flux-pts')
+
+            # Quadrature rule (default to that of the solution points)
+            qrule = self.cfg.get(self.cfgsect, f'quad-pts-{itype}', rname)
+            try:
+                qdeg = self.cfg.getint(self.cfgsect, f'quad-deg-{itype}')
+            except NoOptionError:
+                qdeg = self.cfg.getint(self.cfgsect, 'quad-deg')
+
+            q = get_quadrule(itype, qrule, qdeg=qdeg, flags='s')
+            ppts, pwts = _proj_pts(proj, q.pts), q.wts
+            self._surf_quad[itype, proj] = (ppts, pwts)
+        return ppts, pwts
+
+    @staticmethod
+    def _plocs(eles, eidx, pts):
+        nspts, neles, ndims = eles.nspts, len(eidx), eles.ndims
+
+        # Construct the physical location operator matrix
+        plocop = eles.basis.sbasis.nodal_basis_at(pts)
+
+        # Apply the operator to the mesh elements and reshape
+        e_surf = eles.eles[:,eidx]
+        ploc = plocop @ e_surf.reshape(nspts, -1)
+        ploc = ploc.reshape(len(pts), neles, ndims)
+
+        return ploc
+
+    @staticmethod
+    def _jacs(eles, eidx, pts):
+        neles, ndims = eles.neles, eles.ndims
+
+        smats, djacs = eles._smats_djacs_mpts
```

Comment:
Avoid accessing private members (or refactor if we have to so they are no longer private).